openapi/openapi_v2_backend_configuration.go

```diff
@@ -34,6 +38,66 @@ func (o specV2BackendConfiguration) getHost() (string, error) {
 	return o.spec.Host, nil
 }
 
+func (o specV2BackendConfiguration) getHostByRegion(region string) (string, error) {
+	if region == "" {
+		return "", fmt.Errorf("can't get host by region, missing region value")
+	}
+	isMultiRegion, host, allowedRegions, err := o.isMultiRegion()
+	if err != nil {
+		return "", err
+	}
+	if !isMultiRegion {
+		return "", fmt.Errorf("missing '%s' extension or value provided not matching multiregion host format", extTfProviderMultiRegionFQDN)
+	}
+	if err := o.validateRegion(region, allowedRegions); err != nil {
+		return "", err
+	}
+	overrideHost, err := openapiutils.GetMultiRegionHost(host, region)
+	if err != nil {
+		return "", err
+	}
+	return overrideHost, nil
+}
+
+func (o specV2BackendConfiguration) validateRegion(region string, allowedRegions []string) error {
+	for _, r := range allowedRegions {
+		if r == region {
+			return nil
+		}
+	}
+	return fmt.Errorf("region %s not matching allowed ones %+v", region, allowedRegions)
+}
+
+func (o specV2BackendConfiguration) getDefaultRegion() (string, error) {
+	isMultiRegion, _, regions, err := o.isMultiRegion()
```

Comment:
if `getDefaultRegion` is always called after `isMultiRegion` maybe consider accepting regions as a parameter?

Reply:
+1, done